cmd/pbstatuser/main.go

```diff
@@ -106,15 +106,20 @@ func checkSourceAvailabilityAzure(ctx context.Context) {
 	defer processingWG.Done()
 
 	for s := range chAzure {
-		sr := kafka.SourceResult{
-			SourceID:     s.SourceID,
-			Identity:     s.Identity,
-			ResourceType: "Source",
-		}
-		// TODO: check if source is avavliable - WIP
-		sr.Status = kafka.StatusAvaliable
-		chSend <- sr
-		metrics.IncTotalAvailablilityCheckReqs(models.ProviderTypeAzure, "statuser", sr.Status, nil)
+		metrics.ObserveAvailablilityCheckReqsDuration(models.ProviderTypeAzure, func() error {
+			var err error
+			sr := kafka.SourceResult{
+				SourceID:     s.SourceID,
+				Identity:     s.Identity,
+				ResourceType: "Source",
+			}
+			// TODO: check if source is avavliable - WIP
+			sr.Status = kafka.StatusAvaliable
+			chSend <- sr
+			metrics.IncTotalAvailabilityCheckReqs(models.ProviderTypeAzure, "statuser", sr.Status, nil)
+
+			return fmt.Errorf("An error has occured %w:", err)
```

Comment:
```suggestion
			return fmt.Errorf("an error has occurred %w:", err)
```

Tho I would suggest something more descriptive: `error during check`